lib/Cron/MaintenanceHeavy.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+
+/**
+ * Circles - Bring cloud-users closer together.
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Maxence Lange <maxence@artificial-owl.com>
+ * @copyright 2017
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+namespace OCA\Circles\Cron;
+
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJob;
+use OCP\BackgroundJob\TimedJob;
+use OCA\Circles\Service\MaintenanceService;
+
+class MaintenanceHeavy extends TimedJob {
+	private MaintenanceService $maintenanceService;
+
+
+	/**
+	 * @param ITimeFactory $time
+	 * @param MaintenanceService $maintenanceService
+	 */
+	public function __construct(ITimeFactory $time, MaintenanceService $maintenanceService) {
+		parent::__construct($time);
+
+		$this->setInterval(10);
```

Comment:
This needs to be every day, otherwise they are not really time insensitive ;)

Reply:
fixed. Also removed the night-time check before running heavy process, as it should be managed by the INSENSITIVE flag